### PR TITLE
OLS-136: Handle data collector errors

### PR DESCRIPTION
**Pull request by onmete**
## Description

Handling of errors for fet k8s objects.

## Type of change

- [x] Refactor

## Related

- [OLS-486](https://issues.redhat.com/browse/OLS-486)

### CODE REVIEW

**Comment by onmete** [comment]
@bparees Missing cluster_id is not blocking posting to ingress nor breaking ccx side.
Archive posts with "invalid" cluster_id will still land in our bucket.
Cluster id is useful for tracking the request, but is not critical at the moment.
Hence I choose to not fail completely in case when we are not able to get the cluster id.

File: ols/user_data_collection/data_collector.py [hunk 1]
```diff
@@ -243,24 +258,20 @@ def upload_data_to_ingress(tarball: io.BytesIO) -> requests.Response:
...
-
-    with requests.Session() as s:
-        s.headers = {
-            # TODO: OLS-486 User agent needs to be set as insights operator,
-            # otherwise JWT token is not accepted - we use just random hash
-            # "User-Agent": f"openshift-lightspeed-operator/user-data-collection cluster/{cluster_id}",  # noqa E501
-            "User-Agent": f"insights-operator/abcdefgijklmnopqrstuvwxyz01234567891011 cluster/{cluster_id}",  # noqa: E501
+        token = get_cloud_openshift_pull_secret()
+        headers = {
+            "User-Agent": f"openshift-lightspeed-operator/user-data-collection cluster/{cluster_id}",  # noqa E501
```
Comment (tisnik) [inline ols/user_data_collection/data_collector.py:269]: I'd probably put it into named constant

**Review by tisnik — COMMENTED** [review]
looks good, just have one nitpick (the identifier)

**Pull request merged**